src/main/java/edu/ucla/library/iiif/auth/services/AccessCookieCryptoService.java

```diff
@@ -27,30 +23,16 @@
      */
     String ADDRESS = AccessCookieCryptoService.class.getName();
 
-    /**
-     * The failure code if the service is configured improperly.
-     */
-    int CONFIGURATION_ERROR = 1;
-
-    /**
-     * The failure code if the service receives a decryption request for a cookie that has been tampered with.
-     */
-    int TAMPERED_COOKIE_ERROR = 2;
-
     /**
      * Creates an instance of the service.
      *
      * @param aConfig A configuration
      * @return The service instance
-     * @throws ServiceException if the service implementation isn't configured properly
+     * @throws GeneralSecurityException if the service implementation isn't configured properly
      */
     @SuppressWarnings({ "PMD.PreserveStackTrace" })
-    static AccessCookieCryptoService create(final JsonObject aConfig) {
-        try {
-            return new AccessCookieCryptoServiceImpl(aConfig);
-        } catch (final InvalidKeySpecException | NoSuchAlgorithmException | NoSuchPaddingException details) {
-            throw new ServiceException(CONFIGURATION_ERROR, details.getMessage());
-        }
+    static AccessCookieCryptoService create(final JsonObject aConfig) throws GeneralSecurityException {
+        return new AccessCookieCryptoServiceImpl(aConfig);
```

Comment:
Why not AccessCookieService and AccessCookieServiceImpl, etc. Is the crypto important enough here to include in the name?

Reply:
Agreed, will rename